@martindurant Are you testing with master? (IIRC, this passes on 1.2.4 but fails on master). I can reproduce the failures on CI with master and fastparquet 0.6.2. 
(Failing Web and Docs Build is the example above, and these are the failing tests for Windows py38/Linux Database
```
FAILED pandas/tests/io/test_fsspec.py::test_fastparquet_options - ValueError:...
FAILED pandas/tests/io/test_fsspec.py::test_s3_parquet - ValueError: Opening ...
FAILED pandas/tests/io/test_parquet.py::test_cross_engine_pa_fp - TypeError: ...
FAILED pandas/tests/io/test_parquet.py::TestParquetFastParquet::test_basic - ...
FAILED pandas/tests/io/test_parquet.py::TestParquetFastParquet::test_s3_roundtrip
```
)

--- a/pandas/io/parquet.py
+++ b/pandas/io/parquet.py
@@ -327,9 +327,14 @@ class FastParquetImpl(BaseImpl):
         if is_fsspec_url(path):
             fsspec = import_optional_dependency("fsspec")
 
-            parquet_kwargs["open_with"] = lambda path, _: fsspec.open(
-                path, "rb", **(storage_options or {})
-            ).open()
+            if Version(self.api.__version__) > Version("0.6.1"):
+                parquet_kwargs["fs"] = fsspec.open(
+                    path, "rb", **(storage_options or {})
+                ).fs
+            else:
+                parquet_kwargs["open_with"] = lambda path, _: fsspec.open(
+                    path, "rb", **(storage_options or {})
+                ).open()
         elif isinstance(path, str) and not os.path.isdir(path):
             # use get_handle only when we are very certain that it is not a directory
             # fsspec resources can also point to directories

